vncdotool/loggingproxy.py

```diff
@@ -31,19 +32,20 @@ def connectionMade(self):
         Protocol.connectionMade(self)
         self.transport.setTcpNoDelay(True)
 
-        self.buffer = ''
+        self.buffer = b''
         self.nbytes = 0
         # XXX send version message
         self._handler = self._handle_version, 12
 
     def dataReceived(self, data):
-        self.buffer += data
+        self.buffer = self.buffer + data
         while len(self.buffer) >= self._handler[1]:
             self._handler[0]()
 
     def _handle_version(self):
-        msg = self.buffer[:12]
+        msg = self.buffer[:12].decode('utf-8')
```

Comment:
A hostile server may send a non-UTF-8 sequence and your client will crash than. Instead of converting `bytes` to `str` compare as `bytes` directly as already implemented in c1b89106